File: src/main/java/org/fountainmc/api/event/server/ServerStartEvent.java
package org.fountainmc.api.event.server;

import javax.annotation.Nonnull;
import javax.annotation.ParametersAreNonnullByDefault;

import org.fountainmc.api.Server;
import org.fountainmc.api.event.Event;

import static com.google.common.base.Preconditions.checkNotNull;

@ParametersAreNonnullByDefault
public interface ServerStartEvent extends Event {

    @Nonnull
    Server getServer();

    static ServerStartEvent create(Server server) {
        checkNotNull(server, "Null server");
        return () -> server;
    }

}

File: src/main/kotlin/org/fountainmc/api/event/server/ServerStartEvent.kt
package org.fountainmc.api.event.server

import org.fountainmc.api.Server
import org.fountainmc.api.event.Event

interface ServerStartEvent : Event {

    val server: Server

    companion object {
        fun create(server: Server): ServerStartEvent {
            class SimpleServerStartEvent(override val server: Server): ServerStartEvent
            return SimpleServerStartEvent(server)
        }
    }
}
